rpmio/rpmkeyring.cc

```diff
@@ -61,6 +68,53 @@ rpmKeyring rpmKeyringFree(rpmKeyring keyring)
     return NULL;
 }
 
+rpmKeyringIterator rpmKeyringInitIterator(rpmKeyring keyring, int unused)
+{
+    if (!keyring || unused != 0)
+	return NULL;
+
+    return new rpmKeyringIterator_s {
+	rpmKeyringLink(keyring),
+	new rdlock(keyring->mutex),
```

Comment:
Like said, don't *allocate* this lock, just make it a regular member of the iterator struct. Then you don't need to manually free it.

Reply:
Done. It still needs to be unlocked manually due to the locking madness until #3370 is merged.

Reply:
Oh, that's ... an interesting wrinkle. Ack.